src/include/pbs_nodes.h

```diff
@@ -265,7 +266,7 @@ struct	pbsnode {
 	long			 nd_nsnfree;	/* number of VPs free */
 	long			 nd_ncpus;	/* number of phy cpus on node */
 	short			 nd_written;	/* written to nodes file */
-	unsigned short		 nd_state;	/* state of node */
+	unsigned long		 nd_state;	/* state of node */
```

Comment:
We don't need a long as the state variable, but I wanted to avoid this issue again.  There are other flags that use up bits other than just states.  Right now the highest bit is actually larger than a short can hold, but we're fine because that bit isn't a state.

I just looked at the schema and state is an INTEGER.  Is this specifically an int or is it just a number?

As for your suggested test, I already have one that does just that :)